app/src/services/market_maker.ts

```diff
@@ -139,6 +143,33 @@ class MarketMakerService {
       logger.debug(`Balance information :: Balance ${balance.toString()}`)
       balances.push(balance)
     }
+    return balances
+  }
+  getBalanceInformationByBlock = async (
+    ownerAddress: any,
+    outcomeQuantity: number,
+    blockNumber: number,
+  ): Promise<BigNumber[]> => {
+    const conditionId = await this.getConditionId()
+    const collateralTokenAddress = await this.getCollateralToken()
+
+    const balances = []
+    logger.debug(`Balance information :: Outcomes quantity ${outcomeQuantity}`)
+    for (let i = 0; i < outcomeQuantity; i++) {
+      const collectionId = await this.conditionalTokens.getCollectionIdForOutcome(conditionId, 1 << i)
+      logger.debug(
+        `Balance information :: Collection ID for outcome index ${i} and condition id ${conditionId} : ${collectionId}`,
+      )
+      const positionIdForCollectionId = await this.conditionalTokens.getPositionId(collateralTokenAddress, collectionId)
+
+      const balance = await this.conditionalTokens.getBalanceOfByBlock(
+        ownerAddress,
+        positionIdForCollectionId,
+        blockNumber,
+      )
```

Comment:
@KadenZipfel hey men this is the problematic function that doesn't work on xDai.